### ISSUE TITLE
False negative Wmissing-noreturn warning when a function always throws

### ISSUE THREAD

**Issue by TiborGY** [llvm/llvm-project#146223]
Unlike GCC, Clang's implementation of -Wmissing-noreturn does not fire when seeing a function that always throws:
```
void throwError(const std::string& msg){
	throw std::runtime_error(msg);
}
```
GCC 15.1 issues the following warning:
`warning: function might be candidate for attribute 'noreturn' [-Wsuggest-attribute=noreturn]`

https://godbolt.org/z/nWPjxb377

**Comment by snarang181** [comment]
Created https://github.com/llvm/llvm-project/pull/146234.

**Comment by TiborGY** [comment]
Confirmed fixed
